di/direction.go

```diff
@@ -1,10 +1,46 @@
 package di
 
-type Direction int
+// Direction indicates the layout direction of a piece of text.
+type Direction uint8
 
 const (
+	// DirectionLTR is for Left-to-Right text.
 	DirectionLTR Direction = iota
+	// DirectionRTL is for Right-to-Left text.
 	DirectionRTL
+	// DirectionTTB is for Top-to-Bottom text.
+	DirectionTTB
+	// DirectionBTT is for Bottom-to-Top text.
+	DirectionBTT
+)
+
+// IsVertical returns the layout axis for d. The return value can be
+// used as a value of type axis or as a boolean representing
+// whether the text is vertical.
+//
+//     if d.IsVertical() {
+//     } else {
+//         // Must be Horizontal.
+//     }
+//
+//     switch d.IsVertical() {
+//         case Vertical:
+//         case Horzontal:
+//     }
+//
+func (d Direction) IsVertical() Axis {
```

Comment:
This all looks great about from the `Is...` returning not `bool`, it just doesn't seem to read correctly.
I agree with the idea of `Horizontal` and `Vertical` for other cases - but isn't using that type in result of a boolean operation a little strange?

Reply:
I guess it is odd. I've created a separate method that returns the axis, and changed the signature to return boolean.